### PR TITLE
Send reduction concurrently without blocking votes re-propagation

### LINKED ISSUE #1441: Investigate consensus delay when mempool is full
**Describe "Why" this is needed**
We are experiencing block time (**40-50 seconds**) above the average (8seconds throttled) when the mempool is full.
Atm, the block gas limit is set to ~3B, this means that we have 9-10 transfers per block.
In this case, EST calls took up to 4 seconds, and VST/Accept calls took 2-3 secs

Assuming we have different provisioners for each step the following time should be spent in execute/wait StateTransitions:
* Selection - wait for the block produced with EST: 4seconds.
* 1st Reduction - VST: 3 seconds.
* 2nd Reduction - VST: 3 seconds.  
* Accepting a block (after Agreement) - Accept/Finalize: 3 seconds

**Total: 13 seconds**

To these, we should add the network propagation (not relevant for now).

**Additional context**
Enabling the auto propagation flag on kadcast doesn't change the behaviour

### PR DESCRIPTION
**Pull request by goshawk-3**
fixes #1441 

Additionally, another edge case will be addressed out of this PR (with issue https://github.com/dusk-network/dusk-blockchain/issues/1444)

### CODE REVIEW

File: pkg/core/consensus/reduction/firststep/step.go [hunk 1]
```diff
@@ -98,10 +98,15 @@ func (p *Phase) Run(ctx context.Context, queue *consensus.Queue, evChan chan mes
...
-		r.Round, step, &p.selectionResult.Candidate)
-	// }
+	a := reduction.NewAsyncSend(p.Reduction, r.Round, step, &p.selectionResult.Candidate)
+
+	if p.handler.AmMember(r.Round, step) {
+		_ = a.Go(ctx, &wg, evChan, reduction.Republish)
+	} else {
+		if p.handler.AmMember(r.Round, step+1) {
+			_ = a.Go(ctx, &wg, evChan, reduction.ValidateOnly)
+		}
```
Comment (herr-seppia) [inline pkg/core/consensus/reduction/firststep/step.go:108]: I like this trick

File: pkg/core/consensus/reduction/firststep/step.go [hunk 2]
```diff
@@ -94,14 +94,21 @@ func (p *Phase) Run(ctx context.Context, queue *consensus.Queue, evChan chan mes
...
 
 	p.handler = reduction.NewHandler(p.Keys, r.P, r.Seed)
-	// first we send our own Selection
-	if p.handler.AmMember(r.Round, step) {
-		m, _ := p.SendReduction(r.Round, step, &p.selectionResult.Candidate)
 
-		// Queue my own vote to be registered locally
-		evChan <- m
+	// send our own Selection
+	var wg sync.WaitGroup
```
Comment (herr-seppia) [inline pkg/core/consensus/reduction/firststep/step.go:99]: Any reason to use a `WaitGroup` without waiting for it?

File: pkg/core/consensus/reduction/secondstep/step.go [hunk 3]
```diff
@@ -96,11 +96,15 @@ func (p *Phase) Run(ctx context.Context, queue *consensus.Queue, evChan chan mes
 	p.handler = reduction.NewHandler(p.Keys, r.P, r.Seed)
 	// first we send our own Selection
 
-	if p.handler.AmMember(r.Round, step) {
-		m, _ := p.SendReduction(r.Round, step, p.firstStepVotesMsg.Candidate)
+	// wait group
+	var wg sync.WaitGroup
```
Comment (herr-seppia) [inline pkg/core/consensus/reduction/secondstep/step.go:100]: Any reason to use a WaitGroup without waiting for it?

File: pkg/core/chain/chain.go [hunk 4]
```diff
@@ -384,11 +386,13 @@ func (c *Chain) ProcessSyncTimerExpired(strPeerAddr string) error {
 // acceptSuccessiveBlock will accept a block which directly follows the chain
 // tip, and advertises it to the node's peers.
 func (c *Chain) acceptSuccessiveBlock(blk block.Block, kadcastHeight byte) error {
-	startTime := time.Now().UnixMilli()
-
 	log.WithField("height", blk.Header.Height).Trace("accepting succeeding block")
 
-	prevBlockTimestamp := c.tip.Header.Timestamp
+	// TODO: Verify Certificate
```
Comment (herr-seppia) [inline pkg/core/chain/chain.go:391]: Is this a leftover? If not, a follow up issue would fit the case

File: pkg/core/consensus/reduction/firststep/step.go [hunk 5]
```diff
@@ -94,14 +94,21 @@ func (p *Phase) Run(ctx context.Context, queue *consensus.Queue, evChan chan mes
...
+
+	a := reduction.NewAsyncSend(p.Reduction, r.Round, step, &p.selectionResult.Candidate)
+
+	if p.handler.AmMember(r.Round, step) {
+		_ = a.Go(ctx, &wg, reductionChan, reduction.Republish)
```
Comment (herr-seppia) [inline pkg/core/consensus/reduction/firststep/step.go:104]: Shall the returned cancel function be deferred?

**Review by herr-seppia — COMMENTED** [review]
Not sure about some code snippets

File: pkg/core/consensus/reduction/firststep/step.go (hunk 5 above)
Comment (goshawk-3) [reply pkg/core/consensus/reduction/firststep/step.go:104]: In 1th_reduction step, It is not deferred on purpose.

File: pkg/core/chain/chain.go (hunk 4 above)
Comment (goshawk-3) [reply pkg/core/chain/chain.go:391]: Good catch. I'll file it

File: pkg/core/consensus/reduction/firststep/step.go (hunk 2 above)
Comment (goshawk-3) [reply pkg/core/consensus/reduction/firststep/step.go:99]: It's part of the input params for  AsyncSend.Go, but not using it anymore. So, I'll remove them.

File: pkg/core/chain/chain.go (hunk 4 above)
Comment (goshawk-3) [reply pkg/core/chain/chain.go:391]: FYI https://github.com/dusk-network/dusk-blockchain/issues/1446

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**